erpnext/erpnext_integrations/doctype/affirm_settings/affirm_settings.py

```diff
@@ -223,8 +223,26 @@ def authorize_payment(affirm_data, reference_doctype, reference_docname, integra
 	"""
 		once callback return checkout token it will authroized payment status as failed or sucessful
 	"""
+	payment_request = frappe.get_doc(reference_doctype, reference_docname)
 	redirect_url = "/integrations/payment-failed"
 
+	#if the payment request generated from a quotation, we convert quotation to sale order
+	if payment_request.reference_doctype == "Quotation" and not affirm_data.get("type") == "invalid_request":
+		
+		#we get the quotation document and submit it
+		quotation = frappe.get_doc(payment_request.reference_doctype, payment_request.reference_name)
+		quotation.save(ignore_permissions=True)
```

Comment:
no need to save the quotation here before submitting as you haven't changed any doc values

Reply:
the save is done because the system throws permission errors if I try to submit directly. I can't pass ignore_permissions in .submit() so I'm doing a .save() before a submit.